gcc/rust/ast/rust-expr.h

```diff
@@ -4377,7 +4377,7 @@ class IfLetExpr : public ExprWithBlock
 
   IfLetExpr (std::vector<std::unique_ptr<Pattern> > match_arm_patterns,
 	     std::unique_ptr<Expr> value, std::unique_ptr<BlockExpr> if_block,
-	     std::vector<Attribute> outer_attrs, Location locus)
+	     std::vector<Attribute> /* outer_attrs */, Location locus)
```

Comment:
`IfLetExpr` does have `outer_attrs`. The actual issue here is that the primary constructor does not initialise the `outer_attrs` field from the `outer_attrs` parameter.

Reply:
I'll argue that my change is "correct" -- insofar that it doesn't alter current behavior.  ;-P

---

But if you're saying that there's an actual but here, I'm of course fine if we resolve that one first.  (Thus alter current behavior, with test cases added.)

Reply:
How about we create an issue out of this for now? and then this can be merged in the mean time.

Reply:
I have created a new issue for this: https://github.com/Rust-GCC/gccrs/issues/360